3_run_rsync/tasks/create-pod-service.yml

```diff
@@ -113,7 +125,12 @@
   register: rsync_route
   until: " rsync_route.resources is defined
           and rsync_route.resources[0].spec is defined
-          and rsync_route.resources[0].spec.host is defined"
+          and rsync_route.resources[0].spec.host is defined
```

Comment:
Same `dict.get()` here too. Either that, or all `and` keywords can be removed and conditions can be added as a list:
```yml
until:
- rsync_route.resources | length > 0
- rsync_route.resources[0].spec is defined
.
.
```
Ansible, by default, aggregates all conditions in the list with an `and` clause  between them.

Additionally, `resources` array is always defined in `k8s_facts` response. It's just empty when a resource of interest is not found. Therefore, `rsync_route.resources is defined` should really be `rsync_route.resources | length > 0`